ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessor.java

```diff
@@ -145,6 +145,24 @@ protected AbstractBlockProcessor(
       final List<Transaction> transactions,
       final List<BlockHeader> ommers,
       final PrivateMetadataUpdater privateMetadataUpdater) {
+
+    Result res =
+        executeBlock(
+            blockchain, worldState, blockHeader, transactions, ommers, privateMetadataUpdater);
+
+    // persist accumulated changes from executeBlock
+    worldState.persist(blockHeader);
```

Comment:
why you are not just pass a copy of the worldstade . So you won't have to change this class and persist won't impact the main chain (like the duplicateWorldStateAtParent of the AbstractBlockCreator)

Reply:
I am trying to avoid the `persist` that AbstractBlockProcessor.processBlock does which comits to the underlying storage.  
For executeBlock (which we are calling from engine_executePayload rpc endpoint) we want to defer persisting until we have seen `consensusValidated`.

Reply:
ok it's fine